hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -1053,6 +1053,63 @@
     public static final HazelcastProperty SEARCH_DYNAMIC_CONFIG_FIRST
             = new HazelcastProperty("hazelcast.data.search.dynamic.config.first.enabled", false);
 
+    /**
+     * Defines the pending invocation threshold for the Enterprise WAN
+     * replication implementation. Exceeding this threshold on a WAN
+     * consumer member makes the member delaying the WAN acknowledgment,
+     * thus slowing down the WAN publishers on the source side that send
+     * WAN events to the given WAN consumer. Setting this value to
+     * negative turns off the acknowledgement delaying feature.
+     *
+     * @see #WAN_CONSUMER_ACK_DELAY_BACKOFF_INIT_MS
+     * @see #WAN_CONSUMER_ACK_DELAY_BACKOFF_MAX_MS
+     * @see #WAN_CONSUMER_ACK_DELAY_BACKOFF_MULTIPLIER
+     */
+    public static final HazelcastProperty WAN_CONSUMER_INVOCATION_THRESHOLD
+            = new HazelcastProperty("hazelcast.wan.consumer.invocation.threshold", 1000000);
```

Comment:
Is a million not a bit on the high side? If each item is 5KB (not uncommon) then we are at 5GB of inflight invocations.

Reply:
It is. Reduced to 50k.